### PR TITLE
Related items

### PR DESCRIPTION
**Pull request by alibacova**
- [ ] overall layout and styling for product cards
- [ ] overall layout and styling for carousels
- [ ] thumbnails on hover, change main image functionality
- [ ] click handling for product cards, compare buttons, delete buttons
- [ ] your outfit local storage
- [ ] compare modal overall structure

Still doing some work on comparison modal, making the star rating responsive, resizing the add outfit card

### CODE REVIEW

File: client/src/tests/relatedItems/compareModal.test.jsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import {
+  render, screen, cleanup, fireEvent,
+} from '@testing-library/react/pure';
+import userEvent from '@testing-library/user-event';
+// eslint-disable-next-line import/no-unresolved
+import CompareModal from '../../components/relatedItemsAndComparison/CompareModal.jsx';
+
+let compareModal = null;
+
+const product2 = {
```
Comment (JakeDavisCodes) [inline client/src/tests/relatedItems/compareModal.test.jsx:14]: Smart! good job with the tests!

File: client/src/tests/relatedItems/productCardsCarousel.test.jsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  });
+
+  it('should not render prev slider button when the related products load for the first time', async () => {
+    expect(screen.queryByTestId('product-card-prev-button')).not.toBeInTheDocument();
+  });
+
+  it('should render prev slider button when next button was clicked at least once', async () => {
+    act(() => {
+      // eslint-disable-next-line func-names
+      window.HTMLElement.prototype.scrollIntoView = function () {};
```
Comment (JakeDavisCodes) [inline client/src/tests/relatedItems/productCardsCarousel.test.jsx:53]: You went a lot more into depth with testing than I did! I'm curious what the point of this line is.

File: jest.config.js [hunk 3]
```diff
@@ -1,3 +1,195 @@
...
+
+  // Indicates whether each individual test should be reported during the run
+  // verbose: undefined,
+
+  // An array of regexp patterns that are matched against all source file paths before re-running tests in watch mode
+  // watchPathIgnorePatterns: [],
+
+  // Whether to use watchman for file crawling
+  // watchman: true,
 };
```
Comment (JakeDavisCodes) [inline jest.config.js:195]: I'm guessing this all is just a copy and paste config?

**Review by JakeDavisCodes — APPROVED** [review]
Lots of great work! Just too much great work. Make sure to PR more frequently in the future.

File: client/src/components/relatedItemsAndComparison/Carousel.jsx [hunk 4]
```diff
@@ -23,42 +23,57 @@ function Carousel({ relatedList }) {
...
-        ))}
+          )}
+          {items.map((item, idx) => (
+            <ProductCard
+              classname={classname}
+              key={item}
+              ref={ref}
+              productId={item}
+              index={index}
+              idx={idx}
```
[the remark this answers is not in the document]
Comment (alibacova) [reply client/src/components/relatedItemsAndComparison/Carousel.jsx:68]: Good point. I will rename these to make it clearer

File: client/src/components/relatedItemsAndComparison/YourOutfitList.jsx [hunk 5]
```diff
@@ -1,15 +1,44 @@
...
+  }, [outfitList]);
+
+  const addOutfit = () => {
+    if (!outfitList.includes(product.id)) {
+      setOutfitLst([...outfitList, product.id]);
+    }
+  };
+
+  const deleteOutfit = (id) => {
+    setOutfitLst([...outfitList].filter((itemId) => itemId !== id));
```
[the remark this answers is not in the document]
Comment (alibacova) [reply client/src/components/relatedItemsAndComparison/YourOutfitList.jsx:27]: You're are very right and it's an easy fix too. Thanks!

File: client/src/tests/relatedItems/productCardsCarousel.test.jsx (hunk 2 above)
Comment (alibacova) [reply client/src/tests/relatedItems/productCardsCarousel.test.jsx:53]: jsdom doesn't know what scrollIntoView is, that was a way to bring it in. I learned about it from here: https://stackoverflow.com/questions/53271193/typeerror-scrollintoview-is-not-a-function

File: jest.config.js (hunk 3 above)
Comment (alibacova) [reply jest.config.js:195]: I ran jest init based on the seir workshop video and that's what was generated in the process

**Pull request merged**